### PR TITLE
EmblaCarouselTypeがexportされていない

### LINKED ISSUE #2420: EmblaCarouselTypeがexportされていない
### どのパッケージに問題がありますか？

@yamada-ui/carousel

### パッケージのバージョン

2.0.1

### 概要

Carouselコンポーネントのcontrolrefを使用するときに型定義がめんどくさくなってしまう。

### まさか、問題の修正方法をご存知ですか？

はい

### この問題の修正に参加し、修正を含むプルリクエストを作成しますか？

はい

### PR DESCRIPTION
**Pull request by hinataOhyama**
Closes #2420  

## Description

The type was not provided when using controlref.
Exported EmblaCarouselType as CarouselControlRefType.

### CODE REVIEW

File: packages/components/carousel/src/index.ts [hunk 1]
```diff
@@ -6,3 +6,4 @@ export { CarouselControlPrev, CarouselControlNext } from "./carousel-control"
 export type { CarouselControlProps } from "./carousel-control"
 export { CarouselIndicators } from "./carousel-indicators"
 export type { CarouselIndicatorsProps } from "./carousel-indicators"
+export type { CarouselControlRefType } from "./use-carousel"
```
Comment (hirotomoyamada) [inline packages/components/carousel/src/index.ts:9]: Please change `CarouselControlRefType` to `CarouselControl`.

File: packages/components/carousel/src/use-carousel.ts [hunk 2]
```diff
@@ -27,6 +27,7 @@ export type SlidesInViewOptionsType = EmblaOptionsType["inViewThreshold"]
 export type DragHandlerOptionType = EmblaOptionsType["watchDrag"]
 export type ResizeHandlerOptionType = EmblaOptionsType["watchResize"]
 export type SlidesHandlerOptionType = EmblaOptionsType["watchSlides"]
+export type CarouselControlRefType = EmblaCarouselType
```
Comment (hirotomoyamada) [inline packages/components/carousel/src/use-carousel.ts:30]: Please change `CarouselControlRefType` to `CarouselControl`.

**Comment by hirotomoyamada** [comment]
Hi, thanks for the PR.

Replace all `EmblaCarouselType` in the file with `CarouselControl`.

File: packages/components/carousel/src/index.ts (hunk 1 above)
Comment (hinataOhyama) [reply packages/components/carousel/src/index.ts:9]: I understand!
I'll fix it!

**Comment by hinataOhyama** [comment]
Sorry, please re-review!

File: docs/contents/components/data-display/carousel/props.ja.mdx [hunk 3]
```diff
@@ -67,7 +67,7 @@ tab: Props
 <PropsCard
   id="carousel-controlref"
   name="controlRef"
-  type="RefObject<EmblaCarouselType | undefined>"
+  type="RefObject<CarouselControl | undefined>"
```
Comment (hirotomoyamada) [inline docs/contents/components/data-display/carousel/props.ja.mdx:70]: This file is automatically generated and does not need to be changed.

File: docs/contents/components/data-display/carousel/props.mdx [hunk 4]
```diff
@@ -67,7 +67,7 @@ tab: Props
 <PropsCard
   id="carousel-controlref"
   name="controlRef"
-  type="RefObject<EmblaCarouselType | undefined>"
+  type="RefObject<CarouselControl | undefined>"
```
Comment (hirotomoyamada) [inline docs/contents/components/data-display/carousel/props.mdx:70]: This file is automatically generated and does not need to be changed.

File: packages/components/carousel/DOCS.json [hunk 5]
```diff
@@ -39,7 +39,7 @@
       "description": "Props for carousel control element."
     },
     "controlRef": {
-      "type": "RefObject<EmblaCarouselType | undefined>",
+      "type": "RefObject<CarouselControl | undefined>",
```
Comment (hirotomoyamada) [inline packages/components/carousel/DOCS.json:42]: This file is automatically generated and does not need to be changed.

File: docs/contents/components/data-display/carousel/props.mdx (hunk 4 above)
Comment (hinataOhyama) [reply docs/contents/components/data-display/carousel/props.mdx:70]: I understand!
I'll fix it!

**Review by hirotomoyamada — APPROVED** [review]

**Pull request merged**